blocks/monitoring/aggregation.py

```diff
@@ -151,3 +151,41 @@ def get_aggregator(self):
                               (self.storage, tensor.zeros_like(self.storage))],
                           accumulation_updates=[(self.storage, self.variable)],
                           readout_variable=self.storage)
+
+
+@add_metaclass(ABCMeta)
+class MonitoredQuantity(object):
+    """The base class for monitored-quantities.
+
+    To calculate quantities which are not Theano variables.
+    The calculations are done in an iterative procedure for
```

Comment:
> The calculations are done in an iterative procedure for every single batch

This sentence does not make sense to me, and in general this docstring should be rewritten, e.g.

```python
"""
To monitor a non-Theano quanity in Blocks you have to implement this interface for it. 
The initialize method should do this, accumulate method that, and finallly readout is called to 
do this. 

See Also
------------
   :class:`~blocks.monitoring.evaluators.DatasetEvaluator`, 
   :class:`~blocks.extensions.DataStreamMonitoring`

"""
```